src/components/molecules/TablesUserList/TablesUserList.tsx

```diff
@@ -67,84 +56,93 @@ const firestoreUpdate = (doc: string, update: any) => {
     });
 };
 
-const tableNames = () => {
-  return [...Array(TABLES)].map((_, i: number) => nameOfTable(i));
-};
+const defaultTables = [...Array(TABLES)].map((_, i: number) => createTable(i));
 
 const TablesUserList: React.FunctionComponent<PropsType> = ({
-  limit = 60,
-  imageSize = 35,
-  setUserList,
   experienceName,
+  setSeatedAtTable,
+  seatedAtTable,
+  customTables,
+  TableComponent,
 }) => {
-  const [isExpanded, setIsExpanded] = useState(false);
   const [selectedUserProfile, setSelectedUserProfile] = useState<User>();
   const [showLockedMessage, setShowLockedMessage] = useState(false);
   const [showJoinMessage, setShowJoinMessage] = useState(false);
   const [showLeaveMessage, setShowLeaveMessage] = useState(false);
   const [table, setTable] = useState("");
   const [videoRoom, setVideoRoom] = useState("");
 
+  const nameOfVideoRoom = (i: number) => {
+    return `${experienceName}-table${i + 1}`;
+  };
+
   useFirestoreConnect({ collection: "experiences", doc: experienceName });
-  const { user, users, experience } = useSelector((state: any) => ({
+  const { user, users, experience, usersById } = useSelector((state: any) => ({
     user: state.user,
     users: state.firestore.ordered.users,
+    usersById: state.firestore.data.users,
     experience:
       state.firestore.data.experiences &&
       state.firestore.data.experiences[experienceName],
   }));
 
+  useEffect(() => {
+    if (
+      usersById &&
+      user &&
+      usersById[user.uid] &&
+      usersById[user.uid].data &&
+      usersById[user.uid].data[experienceName] &&
+      usersById[user.uid].data[experienceName].table
```

Comment:
Can you replace with `user && usersById?.[user.uid]?.data?.[experienceName]?.table`